Model/User.php

```diff
@@ -170,24 +151,23 @@ public function unserialize($serialized)
     {
         $data = unserialize($serialized);
 
-        if (9 === count($data)) {
-            unset($data[4], $data[6]);
-
-            // add a few extra elements in the array to ensure that we have enough keys when unserializing
-            // older data which does not include all properties.
-            $data = array_merge($data, array_fill(0, 4, null));
+        if (13 === count($data)) {
```

Comment:
done twice ? What do you mean ? The array count ? I don't think it is worth it. Counting an array of max 13 items is not a bottleneck.
And I don't plan to keep the second case around very long, as it is about people updating from an unreleased version, so it is OK if they have to clear sessions at some point IMO (I'm not even sure we should keep it at all)